libraries/api/src/middleware/auth.js

```diff
@@ -192,7 +179,8 @@ const remoteAuthentication = ({ signatureValidator, entry }) => {
     scopeTemplate = new ScopeExpressionTemplate(entry.scopes);
     // Write route parameters into {[param]: ''}
     // if these are valid parameters, then we can parameterize using req.params
-    let [, params] = utils.cleanRouteAndParams(entry.route);
+    let [, params, optionalParams] = utils.cleanRouteAndParams(entry.route);
+    params = params.filter(param => !optionalParams.includes(param));
```

Comment:
Let's amend the comment about this to add something like:

> We can only decide to `useUrlParams` if all params are _required_ params. Otherwise if they are not provided the scope checking will fail. This means all endpoints with optional params that get included in the scope expression must call `req.authorize`.

In addition this means (I think) that we'll be automatically authorizing fewer places potentially so we need to make sure we're calling `req.authorize` everywhere that we have optional params?

Reply:
This is worth checking manually (it should be quick).  However, we'll get failures if `req.authorize` isn't called either manually or automatically.  For example, commenting out the call in `notify.email()` gives an error running that service's tests.  The error is confusing (send is called twice), but that's known.  The error does get raised, which is what we need here.